experimental/rocm/tracing.c

```diff
@@ -10,11 +10,24 @@
 
 #include "experimental/rocm/status_util.h"
 
+#include <dlfcn.h>
```

Comment:
IREE code is almost always cross-platform and any platform-specific code must be guarded. In this case, we have iree_dynamic_library_t that handles this and it should be used instead.